File: Testing/Stdlib Test Code/test_array_methods_out.c
#include <stdio.h>
#include <stdint.h>
#include <string.h>

int array_sum(int* arr, int n);
int array_contains(int* arr, int n, int val);
int main(void);

int array_sum(int* arr, int n)
{
    int total = 0;
    int i = 0;
    while ((i < n)) {
        total = (total + arr[i]);
        i = (i + 1);
    }
    return total;
}

int array_contains(int* arr, int n, int val)
{
    int i = 0;
    while ((i < n)) {
        if ((arr[i] == val)) {
            return 1;
        }
        i = (i + 1);
    }
    return 0;
}

int main(void)
{
    int nums[] = {3, 1, 4, 1, 5};
    int total = array_sum(nums, 5);
    printf("%d\n", total);
    int found = array_contains(nums, 5, 4);
    printf("%d\n", found);
    int nope = array_contains(nums, 5, 9);
    printf("%d\n", nope);
    return 0;
}

File: Testing/Stdlib Test Code/test_array_methods_out.js
function array_sum(arr, n) {
    let total = 0;
    let i = 0;
    while ((i < n)) {
        total = (total + arr[i]);
        i = (i + 1);
    }
    return total;
}

function array_contains(arr, n, val) {
    let i = 0;
    while ((i < n)) {
        if ((arr[i] === val)) {
            return true;
        }
        i = (i + 1);
    }
    return false;
}

function main() {
    let nums = [3, 1, 4, 1, 5];
    let total = array_sum(nums, nums.length);
    console.log(total);
    let found = array_contains(nums, nums.length, 4);
    console.log(found);
    let nope = array_contains(nums, nums.length, 9);
    console.log(nope);
    return 0;
}


main();
